client/src/components/VoteBlock/index.tsx

```diff
@@ -0,0 +1,100 @@
+import { BiX } from '@react-icons/all-files/bi/BiX';
+import classNames from 'classnames/bind';
+import React, { useState, useRef } from 'react';
+
+import style from './style.module.scss';
+
+const cx = classNames.bind(style);
+
+interface Option {
+  id: number;
+  option: string;
+}
+
+const initialOption: Option[] = [
+  { id: 1, option: '짜장면' },
+  {
+    id: 2,
+    option: '짬뽕',
+  },
+];
+
+function VoteBlock() {
+  const [options, setOptions] = useState<Option[]>(initialOption);
+  const [isWriting, setIsWriting] = useState<boolean>(true);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const onRegister = () => {
+    if (!isWriting) return;
+
+    setIsWriting(false);
+  };
+
+  const onAdd = () => {
+    if (!inputRef.current || !inputRef.current.value) return;
+
+    const { value: option } = inputRef.current;
+
+    const lastId = options.at(-1)?.id;
+    const nextId = lastId !== undefined ? lastId + 1 : 0;
+    const newOption: Option = { id: nextId, option };
+
+    setOptions([...options, newOption]);
+
+    inputRef.current.value = '';
+  };
+
+  const onDelete: React.MouseEventHandler<HTMLDivElement> = (e) => {
+    const { id: targetId } = e.currentTarget.closest('li') as HTMLLIElement;
+
+    const filterdOptions = options.filter(
+      (option) => option.id !== Number(targetId),
+    );
+
+    setOptions(filterdOptions);
+  };
+
+  return (
+    <div className={style['vote-container']}>
+      <h3 className={style.title}>투표</h3>
+      <ul>
+        {options.map(({ id, option }, index) => (
+          <li className={style['option-item']} key={id} id={id.toString()}>
+            <div className={style['box-fill']}>{index + 1}</div>
+            <p className={style.option}>{option}</p>
+            {isWriting && (
+              <div
+                className={cx('box-fill', { 'position-right': true })}
+                onClick={onDelete}
```

Comment:
이렇게 하면 `id={id.toString()}` 이렇게 안넣어줘도 되고,
`const { id: targetId } = e.currentTarget.closest('li') as HTMLLIElement;` 이렇게 안받아와도 돼요!

```suggestion
                onClick={() => onDelete(id)}
```

Reply:
이전 PR도 이걸 의미한거였어요!

Reply:
아아 이렇게 바로 해버려도 되네요 
저 생각을 못했던 이유가 뭔가 `handler`를 바로 전달해버리고 싶어서 그랬던것 같아요 ㅋㅋㅋㅋ
저게 더 깔끔하네요

Reply:
이렇게 쓰면 핸들러 내부 로직은 깔끔해지는데 나중에 useCallback 같은거 사용해서 최적화하는게 어려울 수 있어요!
(아이템이 막 백만개일건 아니니까 그냥 저희가 선택하면 돼요)